### PR TITLE
Jetpack connect: Remove unused route

### PR DESCRIPTION
**Pull request by seear**
Remove `/jetpack/connect/choose/:site` route and associated code. Tracks event `calypso_plans_pre_selection`, added in #19967, shows that this route is never used. 

~This PR does not remove the `showFirst` prop, which was only used for this route. I'm in two minds about this. On the one hand this is unused code, but on the other hand it is useful for experimenting with showing the plans component first as in #19963. Thoughts appreciated.~

**Testing**
All jetpack connect routes should function exactly as before, except `/jetpack/connect/choose/:site`, which will no longer do anything.

### CODE REVIEW

**Comment by sirreal** [comment]
`showFirst` first thoughts: _rip it out_. If we need it later, lets put it back or add something better.

**Comment by seear** [comment]
> showFirst first thoughts: rip it out. If we need it later, lets put it back or add something better.

Yeah, makes more sense. I'll do it in this PR.

File: client/jetpack-connect/plans.jsx [hunk 1]
```diff
@@ -145,59 +145,54 @@ class Plans extends Component {
 	}
 
 	autoselectPlan() {
-		if ( ! this.props.showFirst ) {
```
Comment (seear) [inline client/jetpack-connect/plans.jsx]: Unfortunate diff below, but it should just be the formatting changes resulting from the removal if the above `if`.

**Comment by seear** [comment]
`showFirst` prop removed. Could probably do with another quick review.,

File: client/jetpack-connect/plans.jsx (hunk 1 above)
Comment (sirreal) [reply client/jetpack-connect/plans.jsx]: Thanks for the note, that an inflated diff 😵

**Pull request merged**